grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -807,14 +807,35 @@ class StockMovementApiController {
     }
 
     def requisitionStatusCodes = {
-        def options = RequisitionStatus.listOutboundOptions()?.collect {
-            [
-                    id: it.name(),
-                    value: it.name(),
-                    label: "${g.message(code: 'enum.RequisitionStatus.' + it.name())}",
-                    variant: it.variant.name
-            ]
+        boolean hasRequestApprovalActivity = false
+        boolean isRequestsList = params.getBoolean("isRequestsList")
+        if (params.location) {
+            Location location = Location.get(params.location)
+            hasRequestApprovalActivity = location.supports(ActivityCode.APPROVE_REQUEST)
         }
-        render([data: options] as JSON)
+
+        List<RequisitionStatus> statuses = getRequisitionStatusCodesList(hasRequestApprovalActivity, isRequestsList)
+
+        List options = statuses?.collect(mapStatusCodes)
+        List availableStatuses = RequisitionStatus.listAll().collect(mapStatusCodes)
+
+        render([data: options, availableStatuses: availableStatuses] as JSON)
+    }
+
+    List<RequisitionStatus> getRequisitionStatusCodesList(boolean hasRequestApprovalActivity, boolean isRequestsList) {
+        if (isRequestsList) {
+            return hasRequestApprovalActivity ? RequisitionStatus.listRequisitionOptions() : RequisitionStatus.listOutboundOptions()
+        }
+
+        return hasRequestApprovalActivity ? RequisitionStatus.listOutboundOptionsWhenSupportingRequestApproval() : RequisitionStatus.listOutboundOptions()
+    }
```

Comment:
The idea of this function is to use an appropriate list based on the list page that we are currently working on:
1. If we are on the requests list:
   a) If our location supports request approval activity - add waiting for approval and approved
   b) If our location doesn't support request approval - the options should be the same as we had before
2. If we are on the outbound list:
   a) If our location supports request approval activity - add the approved status
   b) If our location doesn't support request approval - the options should be the same as we had before